### PR TITLE
Update food_cubit.dart, food_item.dart, food_repo.dart. Rename main.d…

### PR DESCRIPTION
**Pull request by Alexandr019**
…art to food_page.dart

### CODE REVIEW

File: lib/food_repo.dart [hunk 1]
```diff
@@ -1,19 +1,16 @@
 import 'package:dio/dio.dart';
 import 'package:food/food_item.dart';
 
-
-
 class FoodRepo {
   Future<List<FoodItem>> fetchFoods() async {
-    final json = await Dio().get(
-        'https://yummly2.p.rapidapi.com/feeds/list',
+    final json = await Dio().get('https://yummly2.p.rapidapi.com/feeds/list',
```
Comment (Oliinyk-Artemii) [inline lib/food_repo.dart:6]: it's better to have an instance of Dio class in a separate field that we'll initialize in a constructor of FoodRepo

File: lib/food_cubit.dart [hunk 2]
```diff
@@ -2,7 +2,6 @@ import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:food/food_repo.dart';
 import 'package:food/food_item.dart';
 
-
 class FoodCubit extends Cubit<FoodState> {
   final FoodRepo _foodRepo = FoodRepo();
```
Comment (Oliinyk-Artemii) [inline lib/food_cubit.dart:6]: it's better to initialize the fields through the constructor

File: lib/food_page.dart [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  void initState() {
+    super.initState();
+    setState(() {
+      _foodCubit = FoodCubit()..fetchFoods();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
```
Comment (Oliinyk-Artemii) [inline lib/food_page.dart:31]: let's move MaterialApp and main method in main.dart, here we will have only Scaffold for FoodPage

**Review by Oliinyk-Artemii — CHANGES_REQUESTED** [review]

File: lib/food_repo.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alexandr019) [reply lib/food_repo.dart]: Done

Comment (Alexandr019) [reply lib/food_repo.dart]: Done

File: lib/food_item.dart [hunk 4]
```diff
@@ -1,30 +1,33 @@
...
+        images: json['content']['details']['images'][0]['hostedLargeUrl'] ,
+        rating: json['content']['details']["rating"],
+        totalTime: json['content']['details']["totalTime"],
+      );
 
   static List<FoodItem> fromJsonList(List<dynamic> jsonList) {
     List<FoodItem> list = [];
     for (int i = 0; i < jsonList.length; i++) {
-      if (jsonList[i]['content']['details']['name'] != null) {
+      if (jsonList[i]['content']['details'] != null &&
```
[the remark this answers is not in the document]
Comment (Alexandr019) [reply lib/food_item.dart]: Done

File: lib/food_cubit.dart (hunk 2 above)
Comment (Alexandr019) [reply lib/food_cubit.dart]: Done

File: lib/food_page.dart (hunk 3 above)
Comment (Alexandr019) [reply lib/food_page.dart]: Done

File: lib/project_router.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+import 'package:food/main.dart';
+
+const _home = '/home';
+
+class ProjectRouter {
+  static Route<dynamic> generateRoute(RouteSettings settings) {
+    switch (settings.name) {
+      case _home:
+        return _buildRoute(const MyApp());
```
Comment (Oliinyk-Artemii) [inline lib/project_router.dart:10]: nope, we need FoodPage here

File: lib/main.dart [hunk 6]
```diff
@@ -1,78 +1,33 @@
...
-                          color: Colors.white,
-                          child: Text(''
-                          ),
-                        );
-                      }),
-            );
-          }),
+      initialRoute: Routes.home.name,
+      debugShowCheckedModeBanner: false,
+      routes: {
```
Comment (Oliinyk-Artemii) [inline lib/main.dart:28]: remove routes and insert
`onGenerateRoute: ProjectRouter.generateRoute,`

File: lib/project_router.dart [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+import 'package:food/main.dart';
+
+const _home = '/home';
```
Comment (Oliinyk-Artemii) [inline lib/project_router.dart:4]: and change to food

Comment (Oliinyk-Artemii) [reply lib/project_router.dart:4]: const _food = '/food';

File: lib/project_router.dart (hunk 5 above)
Comment (Alexandr019) [reply lib/project_router.dart]: Done

File: lib/main.dart (hunk 6 above)
Comment (Alexandr019) [reply lib/main.dart]: Done

File: lib/project_router.dart (hunk 7 above)
Comment (Alexandr019) [reply lib/project_router.dart]: Done

**Review by Oliinyk-Artemii — APPROVED** [review]

**Pull request merged**